optuna/integration/catalyst.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any  # NOQA
+
+import optuna
+
+
+try:
+    from catalyst.dl import Callback
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    # CatalystPruningCallback is disabled because Catalyst is not available.
+    _available = False
+    Callback = object
+
+
+class CatalystPruningCallback(Callback):
```

Comment:
I'd like to hear inputs from other developers but I'm wondering if we should start this feature as `experimental` to guard ourselves against near future changes, as `catalyst` is new to the Optuna "ecosystem" and we don't really know what kinds of APIs are expected by Catalyst users?